src/main/java/fi/vm/yti/datamodel/api/v2/mapper/ModelMapper.java

```diff
@@ -240,25 +187,28 @@ public DataModelInfoDTO mapToDataModelDTO(String prefix, Model model, Consumer<R
         datamodelDTO.setInternalNamespaces(internalNamespaces);
         datamodelDTO.setExternalNamespaces(externalNamespaces);
 
-        Set<TerminologyDTO> terminologies = modelResource.listProperties(DCTerms.references).toList().stream().map(ref -> {
-            var graph = ref.getObject().toString();
-            var terminologyModel = jenaService.getTerminology(graph);
-            if (terminologyModel == null) {
-                return new TerminologyDTO(graph);
-            }
-            return TerminologyMapper.mapToTerminologyDTO(graph, terminologyModel);
-        }).collect(Collectors.toSet());
+        var terminologies = MapperUtils.arrayPropertyToSet(modelResource, DCTerms.requires)
+                .stream().filter(val -> val.startsWith("http://uri.suomi.fi/terminology"))
```

Comment:
Constant values for code list and terminology URIs could be used also in this function

Reply:
Changed to constants in newest commit